src/gui/torrentcontentmodel.cpp

```diff
@@ -551,20 +564,47 @@ void TorrentContentModel::setupModelData(const BitTorrent::AbstractFileStorage &
     emit layoutChanged();
 }
 
-void TorrentContentModel::selectAll()
+void TorrentContentModel::notifyModelUpdate(const QModelIndex &index)
 {
-    for (int i = 0; i < m_rootItem->childCount(); ++i)
+    Q_ASSERT(index.isValid());
+
+    const int lastColumnIndex = columnCount(index) - 1;
```

Comment:
I'm not sure it's possible... But couldn't we update only the first column (i.e. the one that contains the checkbox)? This could be more optimal in terms of performance.

Reply:
There is also 'download priority' column that needs to be updated.